### ISSUE TITLE
Feature: detect nonportable shebangs

### ISSUE THREAD

**Issue by asomers** [koalaman/shellcheck#1047]
A very common cause of shell script portability problems is an assumption that the interpreter lives at a certain location.  For example, Linux users often hardcode "#!/bin/bash", but bash is typically installed to "#!/usr/local/bin/bash" on BSD systems.  Portable scripts should therefore use "#!/bin/sh" or "#!/usr/bin/env <SHELL>" instead.

#### For new checks and feature suggestions
- [x] shellcheck.net (i.e. the latest commit) currently gives no useful warnings about this
- [x] I searched through https://github.com/koalaman/shellcheck/issues and didn't find anything related

#### Here's a snippet or screenshot that shows the problem:

```sh
#!/bin/bash
echo "Hello World!"
```

#### Here's what shellcheck currently says:
<nothing>

#### Here's what I wanted or expected to see:
^-- SCXXXX: "/bin/bash" is nonportable.  Prefer "/usr/bin/env/bash"

**Comment by Nightfirecat** [comment]
Duplicate of #112 and #880. I'd advise reading the comments on each of those issues to understand why that is likely not a good idea.

**Comment by asomers** [comment]
Sorry I couldn't find those earlier issues.  I understand their objections, but all of those objections only apply to scripts whose exact target environment is known.  For example, scripts residing in a package built for a particular OS.  ShellCheck is mostly useful further upstream, for software that hasn't yet been packaged.  In such cases, the only appropriate shebang is a portable one or a blank one (to be added by the Makefile).

**Comment by arth1** [comment]
But it is **not** portable.  It relies on (1) /usr/bin/env being present, and (2) that the shell interpreter of choice is in the PATH.  Neither is a given.
From an AIX machine here:
```
$ which bash
which: 0652-141 There is no bash in /usr/bin /usr/local/bin /etc /usr/sbin /usr/ucb /usr/binX11 /sbin ..
```
On this machine, #!/opt/freeware/bin/bash would be the correct shebang, and a warning that this is better done with env would not be helpful.

Another machine has /usr mounted by NFS.  Until it is mounted, /usr/bin is not available, so /usr/bin/env cannot be called.   Replacing the startup scripts to use /usr/bin/env based on a shellcheck recommendation to do so would render the system unbootable.

Anyhow, the security risk of using env noted in #880 should trump any other concerns here, IMNSHO.

**Comment by Gandalf-** [comment]
Should this issue be closed as invalid?
